### DISCUSSION TITLE
Show usage per user

### DISCUSSION THREAD

**Discussion by iamNoah1** [danny-avila/LibreChat#2213] [Feature Requests & Suggestions]
For governance aspects, we want to be able to track which users are using how much ressources of the respective AI model API. Is this maybe already possible or is something like this planned for the future?

**Comment by danny-avila** [comment]
You can by querying the database and the generated transactions.

For example, with the default docker setup, I can enter the mongo container's shell and use mongosh, where I can execute this command

```sh
db.transactions.aggregate([
  {
    $match: {
      "user": ObjectId("652ac880c4102a77fe54c5db"),
    }
  },
  {
    $group: {
      _id: null,
      totalRawAmount: { $sum: "$rawAmount" },
      totalTokenValue: { $sum: "$tokenValue" }
    }
  }
])

### Results
{
  _id: null,
  totalRawAmount: 27627690.5,
  totalTokenValue: -16297726.142
}
```

I'm making a nice interface, and showing data like this is planned, to be available with the Admin Panel I'm working.

Sign up on the official site for the waitlist, I have an update soon: https://librechat.ai

**Reply by iamNoah1** [reply]
Cool, thanks a bunch
